File: 55_Jump_Game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        
        l=len(nums)
        
        i=0
        while l>i:
            i+=nums[i]
            
            if i==(l-1):
                return True
            if nums[i]<=0:
                return False
            
            
        return False
            
            
            
            
        

File: 55_Jump_Game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        
        
        int max_jump=0;
        
        for(int i=0;i<nums.size();++i)
        {
            if(max_jump < i)
                return false;
            if(max_jump>=nums.size()-1)
                return true;
            max_jump=max(max_jump,i+nums[i]);
        }
        return true;
    }
};